### DISCUSSION TITLE
Using a USB3 to M.2 Enclosure

### DISCUSSION THREAD

**Discussion by bcase7090** [SmoothWAN/SmoothWAN#72] [General & Help]
Quick question, I tossed a Quectel 5G modem into a usb enclosure, plugged it in, did dmesg and it see's it "usb 2-1: Product: RM500Q-AE" . what is the best way about going and mounting this device?

**Comment by TalalMash** [comment]
1. Visit `Interface` tab, click `Add New Interface `
2. Select QMI as a protocol and click next/save
3. In device selection, do you see cdc-wdm0?
4. If yes select it, depending on your carrier you may leave other fields empty.
5. In the firewall tab, select the Red zone
6. Save and Apply.

In case the modem is in ethernet mode, you can check in Devices tab if USB0 or other is added upon plugging and refreshing the page. In this case choose DHCP client in protocol instead.

**Reply by bcase7090** [reply]
Sorry for the delay, work got in the way..

on the drop down menu for device it is black, i do not see cdc-wdm0

**Reply by TalalMash** [reply]
Does the `Devices` tab show anything new when the modem is plugged in?

**Comment by bcase7090** [comment]
It does not, it's blank.

**Reply by TalalMash** [reply]
This may be confusing, the device tab is at the top, it cannot be empty:
![image](https://user-images.githubusercontent.com/96490382/195097805-c3db018a-a0f6-4402-b285-76b1140658b4.png)

**Reply by bcase7090** [reply]
Nope, i do not see cdc-wdm0 in the device tab
<img width="1224" alt="Screen Shot 2022-10-15 at 7 21 04 PM" src="https://user-images.githubusercontent.com/100048528/196011045-5bea106e-7f74-4c72-846b-51d54b2a1064.png">

**Reply by TalalMash** [reply]
Ok so it seems like usb0 or usb1 should be added with DHCP protocol.

**Reply by bcase7090** [reply]
Gotcha, would that be under the "Configure" tab?

**Reply by TalalMash** [reply]
it would be under the interfaces tab as the following:

![image](https://user-images.githubusercontent.com/96490382/196029584-11572d40-10c4-4b12-b2ab-c0eb68cd41fa.png)
---
![image](https://user-images.githubusercontent.com/96490382/196029595-a54c683d-d037-4e86-ad2b-575829582cff.png)
---
![image](https://user-images.githubusercontent.com/96490382/196029616-22a2fe0a-4982-4651-b3b9-8d13e3c8a656.png)
---
![image](https://user-images.githubusercontent.com/96490382/196029638-2c9bdf6a-9c65-4fed-b6aa-fc311a72d995.png)
---
Then Save and Apply.

**Reply by bcase7090** [reply]
Would the protocol be QMI Cellular since it's a qualcomm modem? that way it would allow me to input the APN for my service priovider

**Reply by TalalMash** [reply]
`usb*` generally means that it's an ethernet connection. 
You would have to plug in the dongle to a computer to change the mode to QMI by sending AT commands over the serial port.
e.g `AT+QCFG="usbnet",0 `

**Reply by TalalMash** [reply]
To add, ethernet mode is more reliable in Multi-WAN configuration on OpenWRT, next version will have a modified QMI initialization script.

**Reply by bcase7090** [reply]
gotcha, so would it be better to run quectel  modem in a "router" then lan out to a USB network cable adapter into the pi then?

**Reply by TalalMash** [reply]
I may have confused the terms here, ethernet mode as in ethernet via USB mode, the modem would look like a USB to Ethernet adapter to host/Pi, and the stick doing the routing and configuration.
Simply choose DHCP mode and you're all set.
If you need to change APN or other settings, you'll need configure the modem via AT on a computer.
In QMI mode, you would be changing the APN in OpenWRT instead.

**Reply by bcase7090** [reply]
ah now i understand

**Reply by bcase7090** [reply]
It is set right now to QMI mode, the modem was pulled out of router that had openwrt running on it with a m.2 slot built in. I guess i just need to serial port in, change it to ethernet mode and set the APN via at commands.

**Reply by bcase7090** [reply]
I got it switched over to ECM mode, i pushed the APN command, confirmed on anther device that the USB adapter is working. pluged it into the pi. Created a new interface, set the DHCP, device was usb0, and did the firewall settings, saved and rebooted, still nothing. i did confirm in anther device that it was doing DHCP and it did get "HostlessModem #1 Connected with IP 192.168.225.1"

**Reply by bcase7090** [reply]
NVM on that, i needed to set it to USB1, not usb0

**Reply by bcase7090** [reply]
just thinking ahead, lets say i have 2 of these devices, and they are both the same, if by default they both do a 192.168.255.X ip address, would that cause a conflict?

**Reply by TalalMash** [reply]
> just thinking ahead, lets say i have 2 of these devices, and they are both the same, if by default they both do a 192.168.255.X ip address, would that cause a conflict?

It should not cause any issues as Speedily filters by device not the source IP.
Same with VPN Bypass.
